features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/ecp/ecp_internal_alt.c

```diff
@@ -222,26 +232,31 @@ unsigned char mbedtls_internal_ecp_grp_capable( const mbedtls_ecp_group *grp )
 
 int mbedtls_internal_ecp_init( const mbedtls_ecp_group *grp )
 {
-    /* TODO: Change busy-wait with other means to release CPU */
-    /* Acquire ownership of ECC accelerator */
-    while (! crypto_ecc_acquire());
-    
-    /* Init crypto module */
-    crypto_init();
-    ECC_ENABLE_INT();
-
+    /* Address mbedtls_internal_ecp_init()/mbedtls_internal_ecp_free() are not strictly paired
```

Comment:
Is this change needed any longer? The intent is that the `init` is held only for a short sequence of operations, and your case 2 certainly couldn't happen on a single thread.

Breaking it up like this does make it more granular, I guess - gives the opportunity to interleave ops. Not sure how important that is for ECP, if at all. (It's not like the lighter-weight stuff like SHA, where there will sometimes be time-critical ones).

Reply:
> Even in a single thread application it is legal to have several mbedtls_xxx_context with overlapping lifetimes. For example initialising ctx1 and then initialising ctx2 without freeing ctx1 first should be supported ...

@kjbracey-arm  Per @yanesca commented as above, it is legal to have two mbedtls_xxx_context overlapping in the same thread, so I move locking ecc AC from `mbedtls_internal_ecp_init` to its real operation period.

Reply:
`internal_ecp_init/free` are not contexts (despite the init/free naming being shared with that for contexts). They're specific paired pre- and post-op calls.

See documentation changes made in https://github.com/ARMmbed/mbedtls/pull/2267

Reply:
@kjbracey-arm With `mbedtls_internal_ecp_init`/`mbedtls_internal_ecp_free` guaranteeing paired, non-overlapping, and non-cross-public-function, I move ecc AC init/free back to them.